### ISSUE TITLE
eslint/no-unused-vars false positive

### ISSUE THREAD

**Issue by camc314** [oxc-project/oxc#10994]
https://discord.com/channels/1079625926024900739/1371280939614998588

```
let cancel = () => {}

export function close() {
    cancel = cancel?.()
}
```

appears to error with oxlint, but not with typescript eslint
